ols/user_data_collection/data_collector.py

```diff
@@ -0,0 +1,177 @@
+"""Collect insights and upload it to the Ingress service.
+
+It collects the data on startup and then it waits for the specified interval.
+"""
+
+# HACK for debug launch - ols import
+# import sys
+# from pathlib import Path
+# current_dir = Path(__file__).parent.absolute()
+# sys.path.insert(0, str(current_dir.parent.parent))
+
+import io
+import logging
+import os
+import pathlib
+import tarfile
+import time
+
+import requests
+
+from ols.user_data_collection import auth
+
+OLS_USER_DATA_PATH = os.environ.get("OLS_USER_DATA_PATH", "./user-data/")
+
+# TODO: move these envs to config?
+OLS_USER_DATA_COLLECTION_INTERVAL = int(
+    os.environ.get("UDC_COLLECTION_INTERVAL", 2 * 60 * 60)
+)  # 2 hours in seconds
+
+# available ingress urls are:
+# - https://console.redhat.com/ - production
+# - https://console.stage.redhat.com/ - staging/dev
+# NOTE: Token is not working for stage...
+INGRESS_URL = os.environ.get("INGRESS_URL", "https://console.redhat.com/")
+INGRESS_ENDPOINT = os.environ.get("INGRESS_ENDPOINT", "api/ingress/v1/upload")
```

Comment:
when we say data to ingress. Where does this data actually go to? And how to look at those metrics?

Reply:
it goes to our CCX/observability ceph bucket after making its way through the ccx/observability ingress pipeline.